common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -47,13 +47,20 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
   public static boolean arrayEquals(
       Object leftBase, long leftOffset, Object rightBase, long rightOffset, final long length) {
     int i = 0;
-    while (i <= length - 8) {
-      if (Platform.getLong(leftBase, leftOffset + i) !=
-        Platform.getLong(rightBase, rightOffset + i)) {
-        return false;
-      }
-      i += 8;
-    }
+
+      // This attempts to speed up the memcmp type of operation, but there is no way
+      // to guarantee that the offsets will be on a word boundary in order to use
+      // Platform.getLong
```

Comment:
No matter what, can't we modify this loop to still copy the portions of the region that do begin on a boundary? You can copy 8 bytes at a time for most of the region and bytes at a time at the start and end.